src/Event/AdminEventExtension.php

```diff
@@ -36,89 +37,155 @@ public function __construct(EventDispatcherInterface $eventDispatcher)
 
     public function configureFormFields(FormMapper $form)
     {
-        $this->eventDispatcher->dispatch(
-            'sonata.admin.event.configure.form',
-            new ConfigureEvent($form->getAdmin(), $form, ConfigureEvent::TYPE_FORM)
-        );
+        if ($this->eventDispatcher instanceof ContractsEventDispatcherInterface) {
+            $event = new ConfigureEvent($form->getAdmin(), $form, ConfigureEvent::TYPE_FORM);
+            $eventName = 'sonata.admin.event.configure.form';
```

Comment:
And then copy `$this->eventDispatcher->dispatch($event, $eventName);` here and use a return statement